src/checkqueue.h

```diff
@@ -95,6 +101,7 @@ class CCheckQueue
                         return fRet;
                     }
                     nIdle++;
+                    if (m_stopped) return true; // TODO: This return value is unused.
```

Comment:
Also, shouldn't the `if (m_stopped)` be *after* the condition variable `wait`? Otherwise notifying it will first empty the queue.
Finally, `m_stopped` could be renamed to `m_request_stop` and only briefly set in `StopWorkerThreads` to true and then immediately after to false. No other member functions (like StartWorkerThreads) touch it

Reply:
> Otherwise notifying it will first empty the queue.

Correct. This is done intentionally as I think that worker threads should exit with a clean check queue, no?

Reply:
> Also, shouldn't the `if (m_stopped)` be _after_ the condition variable `wait`?

This wouldn't work for the master thread that could never return.

Reply:
> Finally, `m_stopped` could be renamed to `m_request_stop` and only briefly set in `StopWorkerThreads` to true and then immediately after to false.

How this approach could work when `StopWorkerThreads()` is called during check queue being processed? I think `m_stopped` should keep its state in that case.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18710#issuecomment-619008172).